service/src/main/java/bio/terra/workspace/service/resource/ValidationUtils.java

```diff
@@ -241,6 +241,14 @@ public static void validateResourceName(String name) {
     }
   }
 
+  public static void validateOptionalResourceName(String name) {
```

Comment:
annotate name with @Nullable

Reply:
why is it called validateOptionalResourceName but still throw exception when the resource name is null?

Reply:
If name is null, then `(name != null && !RESOURCE_NAME_VALIDATION_PATTERN.matcher(name).matches())` is false and it will not throw.